### ISSUE TITLE
Creating teal user guides

### ISSUE THREAD

**Issue by lcd2yyz** [insightsengineering/teal#1010]
As `teal` continue to expand in audience, teal app users are becoming more diverse, which may include stakeholders who are not accustomed in using interactive data visualization tools. Therefore, some general "how to use" type of user guides in helping to orient app users, providing an overview of all the possible features, or brief explanation of the UI buttons, will be very valuable resource.

Initial ideas on potential formats of the user guides
- Video tutorial of example teal apps
- Introduction slides, with annotation on key features or main UIs
- Vignette-like documents (eg. no codes, just lots of screenshots with explanations)?

Potential contents to covers
- General layouts of teal app - which section for what
- Teal behavior - eg. filter/subset input data on right-hand-side panel, change settings of analysis/visualization on left-hand-side panel
- Filter panel 
   - all the different filters and what they mean, what behavior to expect (eg. drag slider or enter range for numerical filters)
   - how to use snapshot manager
   - how to read/interpret module-specific filters
- Various features of teal widgets 
- Reporting feature
   - from each module how to add analysis
   - from reporter tab, how to organize contents and general outputs

Feel free to add more ideas or contents of interest

**Comment by chlebowa** [comment]
Many (most?) of the proposed topics seem to be covered by the _Technical Blueprint_ vignettes.

**Comment by lcd2yyz** [comment]
The intended audience of this user guide are not shiny developer or even app developers, so the technical blueprint vignettes are mostly non-relevant here. 
Target audience are the true app users - stakeholders that interact with the app to view analytical results. These users may not be familiar with the input data, may not understand and not interested in learning about the technical blueprint of the product. Rather they just want to be able to use the app to get results. So we need to lower the bar to help them understand where things are, and what each button/UI do. 
Something like [this](https://www.health.wa.gov.au/~/media/Files/Corporate/general-documents/hand-hygiene/PDF/PowerBI-Dashboard-User-Guide.pdf) but hopefully ours can be more elaborate.

**Comment by vedhav** [comment]
@lcd2yyz What do you think about implementing [cicerone](https://cicerone.john-coene.com/) to create guided tours instead of documenting about this extensively?
I see a risk of a teal user getting overwhelmed by all the docs we have if we keep them close to each other. Often times finding these docs is harder and having the docs within the app is a good solution for this. But, I think the bigger question here is that we need to redesign teal to be more user-friendly so extensive user docs are not needed. I believe cicerone is a step in the right direction and hopefully, after some UX design reviews, we can improve it further.

One more argument is that I can still see a completely new user still alienated from teal app features because apart from the common teal app features like the filter panel we will not be able to create extensive docs for all the encodings available in the teal module packages. Cicerone could be extended to modules as well for a better UX and onboard people do actionable things, I can also see a future where we create a widget for this and allow the app developer to modify the help-texts if they want, perhaps specify some study specific information that might be easier for the app user to understand.

**Comment by vedhav** [comment]
I do see benefits of a vignette like "How to use teal" which would be similar to the [getting started vignette](https://insightsengineering.github.io/teal/main/articles/getting-started-with-teal.html) but strictly from a usage perspective.
In the video that I am making I did not dive deep into how to use all the features of teal but it's closely linked with this as you mentioned on Slack. I think the doc form of that video can be made into this vignette. But, I would like this to be very-short and concise like a cheatsheet pdf with lot of graphics and information delivered. Ideally, one feature explained with just one screenshot (with annotations and graphics) for it to be valuable.

**Comment by lcd2yyz** [comment]
> I would like this to be very-short and concise like a cheatsheet pdf with lot of graphics and information delivered. Ideally, one feature explained with just one screenshot (with annotations and graphics) for it to be valuable.

Yes exactly!

> What do you think about implementing [cicerone](https://cicerone.john-coene.com/) to create guided tours

I'm not familiar with the package itself, but it seems very similar to those "WalkMeThrough" widgets, very neat. I've actually implemented something like it for a shiny-dashboard product previously, that we called "app tour", where users can choose to enable or disable the app tour as part of help function (I can share more info via Slack as it's an internal app). 

This looks to be a very promising long-term solution, but obviously will also take more time/efforts to implement at first. So I would consider still create the "how to use" guides/video as a lower-hanging fruit to get the material to users quicker. Once we have a good coverage of the basic material, we can switch gear to develop these types of interactive help features, that can be more robust, comprehensive and easier to maintain.

**Comment by chlebowa** [comment]
@vedhav Note that encodings are part of modules, not `teal` itself. `teal` provides only the module tabs, header/footer and, optionally, the filter panel. Module content is not `teal`. While having a brief explanation of modules would be nice, it should be part of the module, not `teal`. We could possibly build an "exlpain module" feature that will be placed directly in `teal` that will _view_ the explanation for the currently active module.

**Comment by donyunardi** [comment]
To reiterate, the goal of this issue is to create content aimed at app users on how to use the teal app. We can format the content as a PDF, making it easy to edit and share with both internal and external teal app users.

**Comment by kumamiao** [comment]
Please also involve my in this one, will be happy to provide some examples that are more focused on clinical trial usage.

**Comment by donyunardi** [comment]
Considering the UI/UX changes that we're going to apply soon, should we postpone this? @kumamiao

**Comment by chizapoth** [comment]
I'll work on this as part of my onboarding. I think we can create a bit more content on custom modules for new users (on top of the one vignette we had)

This can be combined with teal MDR documentation.

**Comment by kumamiao** [comment]
Thanks a lot Chi @chizapoth . Please kindly note here that the user guide will be for app users rather than app developers. The target audience are app users who may not necessarily know any coding, and create instructions on how to user the app (i.e., navigate filter panels, encoding panel, reporter, etc.).

**Comment by chizapoth** [comment]
Revise this one with example-driven approach; more details on filtering. DDL: within this iteration.

**Comment by chizapoth** [comment]
We are going to make video instead of static guide.

**Comment by donyunardi** [comment]
Here's another idea with [intro.js](https://introjs.com/docs) (20 hours)
 
![Image](https://github.com/user-attachments/assets/c2dd1ec5-0d3f-4fd7-b1a0-113fa8742c9b)

This feels more sophisticated than video content. 

<details>

<summary>Code</summary>

```r
library(teal)
library(shiny)

app <- init(
  data = teal_data(
    iris = iris[1:10, ]
  ),
  modules = modules(
    example_module(label = "my module")
  )
) |>
  modify_title(
    title = "my teal app"
  ) |>
  modify_header(
    tags$head(
      tags$link(
        rel = "stylesheet",
        href = "https://cdnjs.cloudflare.com/ajax/libs/intro.js/8.3.2/intro.min.css"
      ),
      tags$link(
        rel = "stylesheet",
        href = "https://cdnjs.cloudflare.com/ajax/libs/intro.js/8.3.2/introjs.min.css"
      ),
      tags$style(HTML("
      .introjs-tooltip {
        background-color: white !important;
        border: 1px solid #ccc !important;
        border-radius: 5px !important;
        padding: 15px !important;
        box-shadow: 0 4px 6px rgba(0,0,0,0.3) !important;
        min-width: 250px !important;
        max-width: 400px !important;
      }

      .introjs-tooltip {
        z-index: 10000000 !important;
        display: block !important;
        visibility: visible !important;
        opacity: 1 !important;
        position: absolute !important;
      }
      .introjs-overlay {
        z-index: 9999999 !important;
        opacity: 0.8 !important;
      }
      .introjs-helperLayer {
        z-index: 9999998 !important;
      }
      .introjs-tooltipbuttons {
        display: block !important;
      }
      .introjs-button {
        display: inline-block !important;
      }
    ")),
      tags$script(
        src = "https://cdnjs.cloudflare.com/ajax/libs/intro.js/8.3.2/intro.min.js"
      ),
      tags$script(HTML('
      $(document).on("shiny:connected", function() {
        setTimeout(function() {
          console.log("Starting intro.js tour");
          introJs.tour().setOptions({
            steps: [
              {x
                intro: "Welcome to my Teal app! Click Next to continue."
              },
              {
                element: document.querySelector(".teal-navbar"),
                intro: "This is the teal navigation bar."
              },
              {
                element: document.querySelector(".teal-sidebar"),
                intro: "This is the filter panel."
              },
              {
                element: document.querySelector("#teal-teal_modules-nav-my_module-teal_module_ui"),
                intro: "This is the analysis output area."
              }
            ],
            showProgress: true,
            showBullets: false,
            exitOnOverlayClick: false,
            doneLabel: "Done",
            nextLabel: "Next →",
            prevLabel: "← Back"
          }).start();
        }, 2000);
      });
    '))
    )
  )

shinyApp(app$ui, app$server)

```

</details>

**Comment by m7pr** [comment]
@donyunardi @chizapoth 
I think video would be better, than just presentaion.
intro.js looks like it could do a video + looks like you would have some file that you can reuse and update in the future which is ideal as it leaves the reproducible document for the future

**Comment by chizapoth** [comment]
> [@donyunardi](https://github.com/donyunardi) [@chizapoth](https://github.com/chizapoth) I think video would be better, than just presentaion. intro.js looks like it could do a video + looks like you would have some file that you can reuse and update in the future which is ideal as it leaves the reproducible document for the future

I agree, video would be better since there are lots of interactivity involved. 

Another thing might be gif (e.g. the [example in gtsummary](https://www.danieldsjoberg.com/gtsummary/)). This way it's more easy for users to search for the functionality they need.

**Comment by donyunardi** [comment]
We implemented the intro.js [here](https://insightsengineering.github.io/teal/main/articles/getting-started-with-teal.html).
We'll introduce this to the audience for now. 
If it's not enough, we can revisit the video idea.
